common/app-starters-task-launch-request-common/src/main/java/org/springframework/cloud/stream/app/tasklaunchrequest/DataFlowTaskLaunchRequestAutoConfiguration.java

```diff
@@ -77,21 +80,29 @@ public TaskLaunchRequestFunction taskLaunchRequest(TaskLaunchRequestMessageProce
 
 	@Bean
 	@ConditionalOnMissingBean(TaskNameMessageMapper.class)
-	public TaskNameMessageMapper taskNameMessageMapper(DataflowTaskLaunchRequestProperties taskLaunchRequestProperties) {
+	public TaskNameMessageMapper taskNameMessageMapper(DataflowTaskLaunchRequestProperties taskLaunchRequestProperties,
+													   EvaluationContext evaluationContext) {
 		if (StringUtils.hasText(taskLaunchRequestProperties.getTaskNameExpression())) {
 			SpelExpressionParser expressionParser = new SpelExpressionParser();
 			Expression taskNameExpression = expressionParser.parseExpression(taskLaunchRequestProperties.getTaskNameExpression());
-			return new ExpressionEvaluatingTaskNameMessageMapper(taskNameExpression);
+			return new ExpressionEvaluatingTaskNameMessageMapper(taskNameExpression, evaluationContext);
 		}
 
 		return message -> taskLaunchRequestProperties.getTaskName();
 	}
 
+	@Bean
+	public EvaluationContext evaluationContext(BeanFactory beanFactory) {
```

Comment:
This must not be a bean. It is going to collide with existing in Spring Integration.
You may initialize it as a class property in the `@PostConstruct` instead of possible collision.